internal/txapp/txapp.go

```diff
@@ -205,6 +208,33 @@ func (r *TxApp) ChainInfo(ctx context.Context) (int64, []byte, error) {
 	return height, appHash, nil
 }
 
+// ReloadDB reloads the database state into the engine.
+func (r *TxApp) ReloadDB(ctx context.Context) error {
+	tx, err := r.Database.BeginReadTx(ctx)
```

Comment:
Having this here also seems potentially wrong. I am assuming the point of ReloadDB is to reload the in-memory info after syncing data, sync the info is added when the engine constructor is called. Is there a simpler way to manage this?